src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,278 @@
+import { MediaUIAttributes } from './constants.js';
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host ul {
+    list-style: none;
+  }
+</style>
+<ul tabindex="0" role="listbox">
+  <slot></slot>
+</ul>
+`;
+
+class MediaChromeListbox extends window.HTMLElement {
+  #keysSoFar = '';
+  #clearKeysTimeout = null;
+  #slot;
+  #assignedElements;
+
+  static get observedAttributes() {
+    return ['disabled', MediaUIAttributes.MEDIA_CONTROLLER];
+  }
+
+  constructor(options = {}) {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    const listboxHTML = template.content.cloneNode(true);
+    this.nativeEl = listboxHTML;
+
+    shadow.appendChild(listboxHTML);
+
+    this.#slot = this.shadowRoot.querySelector('slot');
+
+    this.#slot.addEventListener('slotchange', () => {
+      this.#assignedElements = this.#slot.assignedElements();
+      const els = this.#items;
+      const activeEls = els.some(el => el.getAttribute('tabindex') === '0');
+
+      // if the user set an element as active, we should use that
+      // rather than assume a default
+      if (activeEls) {
+        return;
+      }
+
+      // default to the aria-selected element if there isn't an
+      // active element already
+      let elToSelect = els.filter(el => el.getAttribute('aria-selected') === 'true')[0];
+
+      // if there isn't an active element or a selected element,
+      // default to the first element
+      if (!elToSelect) {
+        elToSelect = els[0];
+      }
+      
+      elToSelect.setAttribute('tabindex', 0);
+    });
+  }
+
+  get #items() {
+    return this.#assignedElements.filter(el => !el.hasAttribute('disabled'));
```

Comment:
issue(blocking): We can end up with weird edge cases here if `disabled` is set dynamically, especially if the item was also `selected`. Should we consider adding a well-defined event for `disabled`? We could rely on mutation observer(s)? Thoughts

Reply:
So, my thought is that `disabled` should win over "selected". If a user is adding `disabled` they probably should also unselect the item as well. We currently don't have a `selectedElement` getter (but we should add one for #282) but I think it would just return nothing if the selected element ends up disabled.

I'm not quite following why we'd need a mutation observer or a disabled event. Are you saying we should deselect `disabled` items for the user?

Reply:
> Are you saying we should deselect disabled items for the user?

Yes, sorry that was my assumption and also wasn't made clear. If we aren't in a multiselect mode, we can get into weird states if we disable a selected and then re-enable (...undisable?) after having selected another item.

Reply:
Interesting, so, with a `<select>` element, when an option gets disabled, it's still set as the selected item on the select. You just can't re-select it.

So, I think the fix here is that when we reset selected, we don't ignore disabled items. We only ignore selected items for movement.

Reply:
Looks like I already thought of that and forgot to point it out, haha
https://github.com/muxinc/media-chrome/pull/365/files#diff-2b6b4ea5a2bc8a7b804b26adcccf0cc24948c120c635ca3f64e7929241cf4413R178